### PR TITLE
fix: Optional artifacts upload

### PR DESCRIPTION
**Pull request by ls-yannick**
[In my previous PR](https://github.com/leanspace/workflows/pull/24/files) I added a required `AWS_ACCOUNT` secret… but that's a breaking change. The account is required to upload the build artifacts, but I want to roll it our gradually, so here is the same thing in a non-breaking form.

### CODE REVIEW

**Review by jerome-leanspace — COMMENTED** [review]
so you don't want to upload the open api specs anymore?

**Comment by ls-yannick** [comment]
I do, but only if the calling repo passes a spec file and the required AWS Account. I will actually update all the repos to do just that, but I want to be able to do it gradually instead of with a breaking change.
The other option would be to publish the current change under a v2 tag but I feel like this is a more nuclear option, when we have a non-breaking upgrade path that is possible 😁

File: .github/workflows/sonar-nexus-deploy.yml [hunk 1]
```diff
@@ -92,19 +92,16 @@ jobs:
           name: output-jar
           path: ${{ inputs.output-jar-path }}
       # Only upload open api spec artifact if path is set
-      - name: "Upload Open API spec artifact"
```
Comment (jerome-leanspace) [inline .github/workflows/sonar-nexus-deploy.yml:95]: I might be confused but this was removed right?

**Comment by ls-yannick** [comment]
Just running some tests for this over here https://github.com/leanspace/activities-repository/pull/21

File: .github/workflows/sonar-nexus-deploy.yml (hunk 1 above)
Comment (ls-yannick) [reply .github/workflows/sonar-nexus-deploy.yml:95]: Ah, sorry, yes I forgot. Yes it was just there to test that the artifacts were being generated, I should have removed it last time already. This was saving the mon github as build artifacts but we don't need this anymore.

**Review by jerome-leanspace — APPROVED** [review]

**Review by coutinol — APPROVED** [review]

**Pull request merged**